src/pybind/mgr/dashboard/frontend/src/app/shared/components/side-panel/side-panel.component.ts

```diff
@@ -0,0 +1,19 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+
+@Component({
+  selector: 'cd-side-panel',
+  templateUrl: './side-panel.component.html',
+  styleUrl: './side-panel.component.scss'
+})
+export class SidePanelComponent {
+  @Input() expanded = false;
+  @Input() headerText = '';
+  @Input() overlay = true;
```

Comment:
What's the purpose of the `overlay` flag? Seems to me having `expanded` is enough